db/database.go

```diff
@@ -2479,6 +2488,15 @@ func (db *DatabaseContext) StartOnlineProcesses(ctx context.Context) (returnedEr
 	}
 	db.backgroundTasks = append(db.backgroundTasks, bgtSyncTime)
 
+	db.AttachmentMigrationManager = NewAttachmentMigrationManager(db)
+	// if we have collections requiring migration, run the job
+	if len(db.RequireAttachmentMigration) > 0 && !db.BucketSpec.IsWalrusBucket() {
+		err := db.AttachmentMigrationManager.Start(ctx, nil)
+		if err != nil {
+			base.WarnfCtx(ctx, "Error trying to migrate attachments for %s with error: %v", db.Name, err)
```

Comment:
If we can't start the attachment manager, we should fail taking the database online, this is a fatal error. `Start` is asynchronous.

This might be worth logging a specific Info message for because we are taking an explicit action., like "Migrating attachment metadata automatically to Sync Gateway 4.0+ for collections %s"

Reply:
Do you still think this should be done? Given we don't block for going online this migration (like resync) my feeling is we should still take db online but with sufficient warnings for it in the logs.

Reply:
This was my misunderstanding about the requirements of how attachment migration works.

I think we should log that creating a database is starting a background task to perform attachment migration, but this can be as is.